eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/EventPositionTest.java

```diff
@@ -0,0 +1,254 @@
+        skipIfNotRecordMode();
+
+        // Arrange
+        final EventPosition enqueuedTimeEventPosition = EventPosition.fromEnqueuedTime(Instant.EPOCH);
+        final EventData enqueuedTimeEvent = receiveMessageHelper(enqueuedTimeEventPosition, 1).toIterable().iterator().next();
+
+        final EventPosition offsetEventPosition = EventPosition.fromOffset(enqueuedTimeEvent.offset(), false);
+        final EventData offsetEvent = receiveMessageHelper(offsetEventPosition, 1).toIterable().iterator().next();
+
+        // Assert
+        Assert.assertEquals(offsetEvent.sequenceNumber(), enqueuedTimeEvent.sequenceNumber() + 1);
+    }
+
+    /**
+     * Test for receiving sequence number with inclusive filter
+     */
+    @Ignore
+    @Test
+    public void testReceiverSequenceNumberInclusiveFilter() {
+        skipIfNotRecordMode();
+
+        // Arrange
+        final EventPosition enqueuedTimeEventPosition = EventPosition.fromEnqueuedTime(Instant.EPOCH);
+        final EventData enqueuedTimeEvent = receiveMessageHelper(enqueuedTimeEventPosition, 1).toIterable().iterator().next();
+
+        final EventPosition nextSequenceNumberEventPosition = EventPosition.fromSequenceNumber(enqueuedTimeEvent.sequenceNumber(), true);
+        final EventData sequenceNumEvent = receiveMessageHelper(nextSequenceNumberEventPosition, 1).toIterable().iterator().next();
+
+        // Assert
+        Assert.assertEquals(sequenceNumEvent.offset(), enqueuedTimeEvent.offset());
+        Assert.assertEquals(sequenceNumEvent.sequenceNumber(), enqueuedTimeEvent.sequenceNumber());
+    }
+
+    /**
+     * Test for receiving sequence number without inclusive filter
+     */
+    @Ignore
+    @Test
+    public void sequenceNumberNonInclusiveFilterFromEnqueuedTime() {
+        skipIfNotRecordMode();
+
+        // Arrange
+        final EventPosition enqueuedTimeEventPosition = EventPosition.fromEnqueuedTime(Instant.EPOCH);
+        final EventData enqueuedTimeEvent = receiveMessageHelper(enqueuedTimeEventPosition, 1).toIterable().iterator().next();
+
+        final EventPosition nextSequenceNumberEventPosition = EventPosition.fromSequenceNumber(enqueuedTimeEvent.sequenceNumber(), false);
+        final EventData sequenceNumberEvent = receiveMessageHelper(nextSequenceNumberEventPosition, 1).toIterable().iterator().next();
+
+        // Assert
+        Assert.assertEquals(sequenceNumberEvent.sequenceNumber(), enqueuedTimeEvent.sequenceNumber() + 1);
+    }
+
+
+    private Flux<EventData> receiveMessageHelper(EventPosition eventPosition, int numberOfEvents) {
+        receiver = client.createConsumer(getConsumerGroupName(), PARTITION_ID, eventPosition,
+            new EventHubConsumerOptions());
+        final Flux<EventData> receivedData = receiver.receive();
+        // Act
+        StepVerifier.create(receivedData.take(numberOfEvents))
+            .expectNextCount(numberOfEvents)
+            .verifyComplete();
+
+        return receivedData;
```

Comment:
This isn't doing what you think it is. You're creating another .receive() operation when you do this.